WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift

```diff
@@ -426,7 +424,7 @@ private extension ProductFormActionsFactory {
             return (product.upsellIDs.count > 0 || product.crossSellIDs.count > 0)
         // Downloadable files. Only core product types for downloadable files are able to handle downloadable files.
         case .downloadableFiles:
-            return product.downloadable
+            return product.downloadableFiles.isNotEmpty
```

Comment:
We should keep the logic to only show this row when the product is downloadable. When I turn off the switch for Downloadable Product, this row is still available with the 1 file I uploaded, which is confusing:


https://github.com/woocommerce/woocommerce-ios/assets/5533851/7f2629da-d13a-485b-9348-e12cf1d38089

Should we also remove the uploaded file from the download list when turning off the Downloadable switch?

Reply:
> We should keep the logic to only show this row when the product is downloadable. When I turn off the switch for Downloadable Product, this row is still available with the 1 file I uploaded, which is confusing:

Good point, done in 8ec94ad6625711522c2d785ab3884c93477cb5ca

> Should we also remove the uploaded file from the download list when turning off the Downloadable switch?

The backend handles this when saving the product, so we don't need any additional logic for it, that's how Android behaves too.

Reply:
> The backend handles this when saving the product, so we don't need any additional logic for it, that's how Android behaves too.

Oh I mean updating the UI not to show the file there anymore. But if it takes much effort, we can add a GH issue for enhancement (if you think it makes sense).

Reply:
Personally I'll keep it like this, just to follow what the backend does, let's imaging this flow:
1. The user goes to the product settings then toggles the option and confirms the alert.
2. The user toggles the option back without saving the product.

If we keep the logic as it's now, then the files will be restored, same as what the backend and Android does, if we add any logic to remove them, then they won't be restored, WDYT?

Reply:
Sure, that sounds good.